packages/lit-dev-content/samples/examples/motion-todos/my-element.ts

```diff
@@ -0,0 +1,112 @@
+import {TextField} from '@material/mwc-textfield';
+import {Checkbox} from '@material/mwc-checkbox';
+import '@material/mwc-textfield';
+import '@material/mwc-checkbox';
+import '@material/mwc-formfield';
+
+const data = [
+  {id: 1, value: 'Go running.', completed: false},
+  {id: 2, value: 'Strength training', completed: true},
+  {id: 3, value: 'Walk with friends.', completed: false},
+  {id: 4, value: 'Feed the cats.', completed: true},
+  {id: 5, value: 'Shop for dinner.', completed: false},
+  {id: 6, value: 'Order clothes.', completed: false},
+  {id: 7, value: 'BBQ!', completed: false},
+];
+
+type DataItem = typeof data[number];
+
+@customElement('my-element')
+export class MyElement extends LitElement {
+  static styles = styles;
+
+  static shadowRootOptions = {
+    mode: 'open' as ShadowRootMode,
+    delegatesFocus: true,
+  };
+
+  @property({type: Array}) data = data;
+
+  addItem(e: Event) {
+    const nextId = this.data[this.data.length - 1].id + 1;
+    this.data = [
+      ...this.data,
+      {
+        id: nextId,
+        value: (e.target! as TextField).value,
+        completed: false,
+      },
+    ];
+    (e.target! as TextField)!.value = '';
+  }
+
+  removeItem(item: DataItem) {
+    this.data = this.data.filter((i) => i != item);
+  }
+
+  updateItem(updatingItem: DataItem, completed: boolean) {
+    this.data = this.data.map((item) => {
+      if (updatingItem === item) {
+        updatingItem.completed = completed;
+      }
+      return item;
+    });
+  }
+
+  render() {
+    const keyframeOptions = {
+      duration: 500,
+      fill: 'both' as FillMode,
+    };
+    const list = (completed = false) => html`<div
+      class="list ${classMap({completed})}"
+    >
+      <h3>${completed ? `Completed` : `Todos`}</h3>
+      <ul tabindex="-1">
+        ${repeat(
+          this.data.filter((item) =>
+            completed ? item.completed : !item.completed
+          ),
+          (item) => item.id,
+          (item) => html`<li
+            ${animate({
+              keyframeOptions,
+              in: flyBelow,
+              out: fadeOut,
+              stabilizeOut: true,
+              id: `${item.id}:${completed ? 'right' : 'left'}`,
+              inId: `${item.id}:${completed ? 'left' : 'right'}`,
+              skipInitial: true,
+            })}
+          >
+            <mwc-formfield label="${item.id}. ${item.value}"
+              ><mwc-checkbox
+                type="checkbox"
+                ?checked=${completed}
+                @change=${(e: Event) =>
+                  this.updateItem(item, (e.target! as Checkbox).checked)}
+              ></mwc-checkbox></mwc-formfield
+            ><button @click=${() => this.removeItem(item)}>
+              remove_circle_outline
+            </button>
+          </li>`
+        )}
+      </ul>
+    </div>`;
+    return html`
+      <mwc-textfield
+        outlined
+        @change=${this.addItem}
+        label="Enter a todo..."
```

Comment:
Maybe add an "Add" button here? Took me a second to realize I should press Enter to add a new item.

Reply:
Done.